airbyte-integrations/connectors/source-google-sheets/unit_tests/integration/custom_http_mocker.py

```diff
@@ -0,0 +1,73 @@
+# Copyright (c) 2024 Airbyte, Inc., all rights reserved.
+
+
+from functools import wraps
+from typing import Dict
+from unittest.mock import patch
+from urllib.parse import parse_qsl, unquote, urlencode, urlunparse
+
+from airbyte_cdk.test.mock_http import HttpResponse
+from airbyte_cdk.test.mock_http.request import HttpRequest
+from httplib2 import Response
+
+
+def parse_and_transform(parse_result_str: str):
+    """
+    Parse the input string representation of a HttpRequest transform it into the URL.
+    """
+    parse_result_part = parse_result_str.split("ParseResult(", 1)[1].split(")", 1)[0]
+
+    # Convert the ParseResult string into a dictionary
+    components = eval(f"dict({parse_result_part})")
+
+    url = urlunparse((
+        components["scheme"],
+        components["netloc"],
+        components["path"],
+        components["params"],
+        components["query"],
+        components["fragment"],
+    ))
+
+    return url
+
+
+class CustomHttpMocker:
```

Comment:
That's a great idea. I like it a lot. This really show the power on encapsulation: If we eventually expand the idea, we would have an interface where we can have multiple implementation (requests, httplib2, a class that spawns a wiremock HTTP server, etc...) although we can assume this is a one-off for this case.